### ISSUE TITLE
Rule Change: `require-unicode-regexp` supports `v` flag

### ISSUE THREAD

**Issue by sosukesuzuki** [eslint/eslint#17199]
### What rule do you want to change?

`require-unicode-regexp`

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
/^[👍]$/v.test("👍")
```

### What does the rule currently do for this code?

Cannot parse. And even if it could be parsed, it would probably generate a warning.

### What will the rule do after it's changed?

There are two change requests:

- `v` flag is a superset of `u` flag so that this rule will not warn if `v` flag is enabled
- Add an option like `requireVFlag`. When this option is enabled, ESLint warns if `v` flag is disabled, even if `u` flag is enabled.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

- RegExp v flag is a Stage 4  proposal( https://github.com/tc39/proposal-regexp-v-flag ).
- We need to wait https://github.com/eslint-community/regexpp/pull/82 for this.

**Comment by mdjermanovic** [comment]
I agree that this rule should at least not warn if the `v` flag is present, but it might be better to open a [Request for new syntax support](https://github.com/eslint/eslint/issues/new?assignees=&labels=core%2Cnew+syntax&projects=&template=new-syntax.yml) issue and continue the discussion there.

**Comment by sosukesuzuki** [comment]
New issue is opened, so close this. https://github.com/eslint/eslint/issues/17223
